app/src/mainwindow2.cpp

```diff
@@ -417,6 +416,102 @@ void MainWindow2::clearRecentFilesList()
     getPrefDialog()->updateRecentListBtn(!recentFilesList.isEmpty());
 }
 
+void MainWindow2::pegBarReg()
+{
+    if (pegreg != nullptr)
+    {
+        QMessageBox::information(this, nullptr,
+                                 tr("Dialog is already open!"),
+                                 QMessageBox::Ok);
+        return;
+    }
+
+    pegreg = new PegBarAlignmentDialog(this);
+    updatePegRegLayers();
+    connect(mEditor->select(), &SelectionManager::selectionChanged, this, &MainWindow2::updatePegReg);
+    connect(mEditor, &Editor::currentFrameChanged, this, &MainWindow2::updatePegReg);
+    connect(mEditor->layers(), &LayerManager::currentLayerChanged, this, &MainWindow2::updatePegReg);
+    connect(mEditor->layers(), &LayerManager::layerCountChanged, this, &MainWindow2::updatePegRegLayers);
+    connect(pegreg, &PegBarAlignmentDialog::cancelPressed, this, &MainWindow2::closePegReg);
+    connect(pegreg, &PegBarAlignmentDialog::alignPressed, this, &MainWindow2::alignPegs);
+    connect(pegreg, &PegBarAlignmentDialog::layerListClicked, this, &MainWindow2::updatePegReg);
+    pegreg->setRefLayer(mEditor->layers()->currentLayer()->name());
+    pegreg->setRefKey(mEditor->currentFrame());
+    pegreg->setLabRefKey();
+    pegreg->setWindowFlag(Qt::WindowStaysOnTopHint);
+    pegreg->show();
+
+    updatePegReg();
+}
+
+void MainWindow2::closePegReg()
+{
+    disconnect(mEditor->select(), &SelectionManager::selectionChanged, this, &MainWindow2::updatePegReg);
+    disconnect(mEditor, &Editor::currentFrameChanged, this, &MainWindow2::updatePegReg);
+    disconnect(mEditor->layers(), &LayerManager::currentLayerChanged, this, &MainWindow2::updatePegReg);
+    disconnect(mEditor->layers(), &LayerManager::layerCountChanged, this, &MainWindow2::updatePegRegLayers);
+    disconnect(pegreg, &PegBarAlignmentDialog::cancelPressed, this, &MainWindow2::closePegReg);
+    disconnect(pegreg, &PegBarAlignmentDialog::alignPressed, this, &MainWindow2::alignPegs);
+    disconnect(pegreg, &PegBarAlignmentDialog::layerListClicked, this, &MainWindow2::updatePegReg);
+    pegreg->close();
+    pegreg = nullptr;
+}
+
+void MainWindow2::updatePegRegLayers()
```

Comment:
This function should be moved into the pegbar class. If you need to access Editor and related core functionality, add a ```void setCore(Editor*)``` in your header and keep a reference of Editor named "mEditor". Alternatively you can add it as parameter in the constructor.